x-pack/plugins/ml/server/models/data_recognizer/data_recognizer.ts

```diff
@@ -247,6 +252,75 @@ export class DataRecognizer {
     return configs.find((i) => i.module.id === id);
   }
 
+  // called externally by an endpoint
+  public async findIndexMatches(
+    moduleId: string,
+    size: number = 15
+  ): Promise<RecognizeModuleResultDataView[]> {
+    const config = await this._findConfig(moduleId);
+    const matchingDataViews: RecognizeModuleResultDataView[] = [];
+
+    if (config?.module.query === undefined) {
+      return matchingDataViews;
+    }
+
+    try {
+      const promises: Array<
+        Promise<SearchResponse<unknown, Record<string, AggregationsAggregate>>>
+      > = [];
+      // Keep track of which data view title is associated with which promise
+      const dataViewTitleMap: Record<string, Promise<any>> = {};
+      const dataViewsMap: Record<string, { id: string; name: string | undefined }> = {};
```

Comment:
For both of these maps we can't use empty objects created by `{}` any more, due to potential prototype pollution vulnerabilities. We should use a `Map` or create the object with `Object.create(null)`

Reply:
Removed in c95eeaac6e88ebed1292d54ceeea5a6babbc65d1